avahi-core/dns.c

```diff
@@ -802,8 +802,13 @@ uint8_t* avahi_dns_packet_append_record(AvahiDnsPacket *p, AvahiRecord *r, int c
     if (append_rdata(p, r) < 0)
         goto fail;
 
-    size = avahi_dns_packet_extend(p, 0) - start;
-    assert(size <= AVAHI_DNS_RDATA_MAX);
+    end = avahi_dns_packet_extend(p, 0);
+    if (!end)
+	goto fail;
+
+    size = end - start;
+    if (size > AVAHI_DNS_RDATA_MAX)
```

Comment:
I think I'd expect `append_rdata` to return an error and that's probably what that assertion expected too. I'll take try to take a closer look. Either way to judge from https://coveralls.io/builds/62671171/source?filename=avahi-core%2Fdns.c#L784 this function isn't covered by any tests so maybe it would be great to add a test case.